### PR TITLE
Move Sharing to Marketing

### PR DESCRIPTION
**Pull request by belcherj**
#### Changes proposed in this Pull Request

* Removes sharing and creates a marketing item in the sidebar

#### Testing instructions

* Observe marketing item under the tools menu
* Click it observe page loads and clicking between tabs works
* Go directly to /sharing/{ siteName }, observe it redirects to /marketing/connections
* Go directly to /sharing/buttons/{ siteName }, observe it redirects to /marketing/sharing-buttons
* Turn off `"ui/streamlined-nav-drawer": true,` in `config/development.json`
* Restart `npm start`
* observe old menu
* Follow same testing above

### CODE REVIEW

**Comment by belcherj** [comment]
Fixed

**Comment by gravityrail** [comment]
@belcherj It might also be worth doing something for users who are redirected for the first time, e.g. a tooltip pointing at the sidebar item that says "Sharing is now part of our Marketing section. Stay tuned for more marketing tools to supercharge your WordPress website!"

**Comment by belcherj** [comment]
@jancavan please see Dan's comments above.

File: client/my-sites/marketing/controller.js [hunk 1]
```diff
@@ -25,6 +25,16 @@ import {
 } from 'state/sites/selectors';
 import versionCompare from 'lib/version-compare';
 
+export const redirectButtons = context => {
```
Comment (stephanethomas) [inline client/my-sites/marketing/controller.js]: What about renaming `redirectButtons` in to `redirectSharingButtons`? It's obviously longer but it better matches the path and name of the page.

Comment (belcherj) [reply client/my-sites/marketing/controller.js]: done

**Comment by stephanethomas** [comment]
I've updated the routing code to avoid the regular expressions. If you're cool with those changes, you've got my approval to merge this pull request (after giving it a last try).

**Comment by jancavan** [comment]
I don't think there needs to be any sort of notice since this isn't really a place users would repeatedly go to, unlike Posts, Customizer, etc. We should, however, update this in our Support docs.

**Comment by jancavan** [comment]
Checking out the PR now. First thing I noticed was the order:

Screenshot 2019-04-22 10 53 53

Marketing should be above Earn as noted in the post p2-p9j7e4-Gx:

> Below is a mockup of the Marketing hub. Users can get to it from the sidebar under Tools. This will replace Sharing. It will be located above “Earn”, with marketing being the natural first step before users can make money from their site.

**Pull request merged**